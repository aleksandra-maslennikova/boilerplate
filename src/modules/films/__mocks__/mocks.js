export const filmsMock = {
  count: 6,
  results: [
    {
      title: "A New Hope",
      episode_id: 4,
      opening_crawl: "It is a period of civil war",
      director: "George Lucas",
      producer: "Gary Kurtz, Rick McCallum",
      release_date: "1977-05-25",
      characters: ["http://swapi.dev/api/people/1/"],
      planets: ["http://swapi.dev/api/planets/1/"],
      starships: [],
      vehicles: [],
      species: [],
      created: "2014-12-10T14:23:31.880000Z",
      edited: "2014-12-20T19:49:45.256000Z",
      url: "http://swapi.dev/api/films/1/",
    },
  ],
};

export const filmMock = {
  title: "A New Hope",
  episode_id: 4,
  opening_crawl: "It is a period of civil war",
  director: "George Lucas",
  producer: "Gary Kurtz, Rick McCallum",
  release_date: "1977-05-25",
  characters: ["http://swapi.dev/api/people/1/"],
  planets: ["http://swapi.dev/api/planets/1/"],
  starships: [],
  vehicles: [],
  species: [],
  created: "2014-12-10T14:23:31.880000Z",
  edited: "2014-12-20T19:49:45.256000Z",
  url: "http://swapi.dev/api/films/1/",
};
