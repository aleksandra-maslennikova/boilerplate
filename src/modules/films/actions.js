/* Api */
import { api } from "../../api";

/* Types */
import { types } from "./types";

export const filmsActions = Object.freeze({
  /* Sync */
  startFetching: () => {
    return {
      type: types.START_FETCHING_FILMS,
    };
  },
  stopFetching: () => {
    return {
      type: types.STOP_FETCHING_FILMS,
    };
  },
  fill: (payload) => {
    return {
      type: types.FILL_FILMS,
      payload,
    };
  },
  setFetchingError: (error) => {
    return {
      type: types.SET_FETCHING_FILMS_ERROR,
      error: true,
      payload: error,
    };
  },

  /* Async */
  fetchAsync: (page) => async (dispatch) => {
    dispatch(filmsActions.startFetching());

    const response = await api.films.fetch(page);

    if (response.status === 200) {
      const { results, count } = await response.json();

      dispatch(filmsActions.fill({ results, count }));
    } else {
      const error = {
        status: response.status,
      };

      dispatch(filmsActions.setFetchingError(error));
    }
    dispatch(filmsActions.stopFetching());
  },
});

export const filmActions = Object.freeze({
  /* Sync */
  startFetching: () => {
    return {
      type: types.START_FETCHING_FILM,
    };
  },
  stopFetching: () => {
    return {
      type: types.STOP_FETCHING_FILM,
    };
  },
  fillFilmProfile: (payload) => {
    return {
      type: types.FILL_FILM_PROFILE,
      payload,
    };
  },
  setFetchingError: (error) => {
    return {
      type: types.SET_FETCHING_FILM_ERROR,
      error: true,
      payload: error,
    };
  },
  clearFilmProfile: () => {
    return {
      type: types.CLEAR_FILM_PROFILE,
    };
  },

  /* Async */
  fetchAsync: (id) => async (dispatch) => {
    dispatch(filmActions.startFetching());

    const response = await api.film.fetch(id);

    if (response.status === 200) {
      const results = await response.json();

      dispatch(filmActions.fillFilmProfile(results));
    } else {
      const error = {
        status: response.status,
      };

      dispatch(filmActions.setFetchingError(error));
    }
    dispatch(filmActions.stopFetching());
  },
});
