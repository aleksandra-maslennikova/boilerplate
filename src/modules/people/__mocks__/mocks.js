export const peopleMock = {
  count: 82,
  results: [
    {
      name: "Wilhuff Tarkin",
      height: "180",
      mass: "unknown",
      hair_color: "auburn, grey",
      skin_color: "fair",
      eye_color: "blue",
      birth_year: "64BBY",
      gender: "male",
      homeworld: "http://swapi.dev/api/planets/21/",
      films: ["http://swapi.dev/api/films/1/", "http://swapi.dev/api/films/6/"],
      species: [],
      vehicles: [],
      starships: [],
      created: "2014-12-10T16:26:56.138000Z",
      edited: "2014-12-20T21:17:50.330000Z",
      url: "http://swapi.dev/api/people/12/",
    },
  ],
};

export const personMock = {
  name: "Luke Skywalker",
  height: "172",
  mass: "77",
  hair_color: "blond",
  skin_color: "fair",
  eye_color: "blue",
  birth_year: "19BBY",
  gender: "male",
  homeworld: "http://swapi.dev/api/planets/1/",
  films: [
    "http://swapi.dev/api/films/1/",
    "http://swapi.dev/api/films/2/",
    "http://swapi.dev/api/films/3/",
    "http://swapi.dev/api/films/6/",
  ],
  species: [],
  vehicles: [
    "http://swapi.dev/api/vehicles/14/",
    "http://swapi.dev/api/vehicles/30/",
  ],
  starships: [
    "http://swapi.dev/api/starships/12/",
    "http://swapi.dev/api/starships/22/",
  ],
  created: "2014-12-09T13:50:51.644000Z",
  edited: "2014-12-20T21:17:56.891000Z",
  url: "http://swapi.dev/api/people/1/",
};
