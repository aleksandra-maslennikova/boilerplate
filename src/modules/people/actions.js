/* Api */
import { api } from "../../api";

/* Types */
import { types } from "./types";

export const peopleActions = Object.freeze({
  /* Sync */
  startFetching: () => {
    return {
      type: types.START_FETCHING_PEOPLE,
    };
  },
  stopFetching: () => {
    return {
      type: types.STOP_FETCHING_PEOPLE,
    };
  },
  fill: (payload) => {
    return {
      type: types.FILL_PEOPLE,
      payload,
    };
  },
  setFetchingError: (error) => {
    return {
      type: types.SET_FETCHING_PEOPLE_ERROR,
      error: true,
      payload: error,
    };
  },

  /* Async */
  fetchAsync: (page) => async (dispatch) => {
    dispatch(peopleActions.startFetching());

    const response = await api.people.fetch(page);

    if (response.status === 200) {
      const { results, count } = await response.json();

      dispatch(peopleActions.fill({ results, count }));
    } else {
      const error = {
        status: response.status,
      };

      dispatch(peopleActions.setFetchingError(error));
    }
    dispatch(peopleActions.stopFetching());
  },
});

export const personActions = Object.freeze({
  /* Sync */
  startFetching: () => {
    return {
      type: types.START_FETCHING_PERSON,
    };
  },
  stopFetching: () => {
    return {
      type: types.STOP_FETCHING_PERSON,
    };
  },
  fillPersonProfile: (payload) => {
    return {
      type: types.FILL_PERSON_PROFILE,
      payload,
    };
  },
  setFetchingError: (error) => {
    return {
      type: types.SET_FETCHING_PERSON_ERROR,
      error: true,
      payload: error,
    };
  },
  clearPersonProfile: () => {
    return {
      type: types.CLEAR_PERSON_PROFILE,
    };
  },

  /* Async */
  fetchAsync: (id) => async (dispatch) => {
    dispatch(personActions.startFetching());

    const response = await api.person.fetch(id);

    if (response.status === 200) {
      const results = await response.json();

      dispatch(personActions.fillPersonProfile(results));
    } else {
      const error = {
        status: response.status,
      };

      dispatch(personActions.setFetchingError(error));
    }
    dispatch(personActions.stopFetching());
  },
});
