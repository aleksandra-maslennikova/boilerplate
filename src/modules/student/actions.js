import { types } from "./types";

export const studentActions = {
  fillProfile: (profile) => {
    return {
      type: types.FILL_STUDENT_PROFILE,
      payload: profile,
    };
  },
};
