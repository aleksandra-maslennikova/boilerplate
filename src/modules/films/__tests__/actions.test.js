/* Actions */
import { filmsActions, filmActions } from "../actions";

/* Mocks */
import { filmsMock, filmMock } from "../__mocks__/mocks";

describe("Films actions", () => {
  it("should start fetching", () => {
    expect(filmsActions.startFetching()).toMatchSnapshot();
  });
  it("should stop fetching", () => {
    expect(filmsActions.stopFetching()).toMatchSnapshot();
  });
  it("should fetch films", () => {
    expect(filmsActions.fetchAsync()).toMatchSnapshot();
  });
  it("should fill films", () => {
    expect(filmsActions.fill(filmsMock)).toMatchSnapshot();
  });
  it("should set fetching error", () => {
    expect(
      filmsActions.setFetchingError("We have an error !")
    ).toMatchSnapshot();
  });
});

describe("Film actions", () => {
  it("should start fetching", () => {
    expect(filmActions.startFetching()).toMatchSnapshot();
  });
  it("should stop fetching", () => {
    expect(filmActions.stopFetching()).toMatchSnapshot();
  });
  it("should fetch person", () => {
    expect(filmActions.fetchAsync()).toMatchSnapshot();
  });
  it("should fill person", () => {
    expect(filmActions.fillFilmProfile(filmMock)).toMatchSnapshot();
  });
  it("should set fetching error", () => {
    expect(
      filmActions.setFetchingError("We have an error !")
    ).toMatchSnapshot();
  });
  it("should clear film profile", () => {
    expect(filmActions.clearFilmProfile()).toMatchSnapshot();
  });
});
