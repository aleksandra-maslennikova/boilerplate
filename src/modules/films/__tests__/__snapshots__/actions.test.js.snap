// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Film actions should clear film profile 1`] = `
Object {
  "type": "CLEAR_FILM_PROFILE",
}
`;

exports[`Film actions should fetch person 1`] = `[Function]`;

exports[`Film actions should fill person 1`] = `
Object {
  "payload": Object {
    "characters": Array [
      "http://swapi.dev/api/people/1/",
    ],
    "created": "2014-12-10T14:23:31.880000Z",
    "director": "George Lucas",
    "edited": "2014-12-20T19:49:45.256000Z",
    "episode_id": 4,
    "opening_crawl": "It is a period of civil war",
    "planets": Array [
      "http://swapi.dev/api/planets/1/",
    ],
    "producer": "Gary Kurtz, Rick McCallum",
    "release_date": "1977-05-25",
    "species": Array [],
    "starships": Array [],
    "title": "A New Hope",
    "url": "http://swapi.dev/api/films/1/",
    "vehicles": Array [],
  },
  "type": "FILL_FILM_PROFILE",
}
`;

exports[`Film actions should set fetching error 1`] = `
Object {
  "error": true,
  "payload": "We have an error !",
  "type": "SET_FETCHING_FILM_ERROR",
}
`;

exports[`Film actions should start fetching 1`] = `
Object {
  "type": "START_FETCHING_FILM",
}
`;

exports[`Film actions should stop fetching 1`] = `
Object {
  "type": "STOP_FETCHING_FILM",
}
`;

exports[`Films actions should fetch films 1`] = `[Function]`;

exports[`Films actions should fetch people 1`] = `[Function]`;

exports[`Films actions should fill films 1`] = `
Object {
  "payload": Object {
    "count": 6,
    "results": Array [
      Object {
        "characters": Array [
          "http://swapi.dev/api/people/1/",
        ],
        "created": "2014-12-10T14:23:31.880000Z",
        "director": "George Lucas",
        "edited": "2014-12-20T19:49:45.256000Z",
        "episode_id": 4,
        "opening_crawl": "It is a period of civil war",
        "planets": Array [
          "http://swapi.dev/api/planets/1/",
        ],
        "producer": "Gary Kurtz, Rick McCallum",
        "release_date": "1977-05-25",
        "species": Array [],
        "starships": Array [],
        "title": "A New Hope",
        "url": "http://swapi.dev/api/films/1/",
        "vehicles": Array [],
      },
    ],
  },
  "type": "FILL_FILMS",
}
`;

exports[`Films actions should fill people 1`] = `
Object {
  "payload": Object {
    "count": 82,
    "results": Array [
      Object {
        "birth_year": "64BBY",
        "created": "2014-12-10T16:26:56.138000Z",
        "edited": "2014-12-20T21:17:50.330000Z",
        "eye_color": "blue",
        "films": Array [
          "http://swapi.dev/api/films/1/",
          "http://swapi.dev/api/films/6/",
        ],
        "gender": "male",
        "hair_color": "auburn, grey",
        "height": "180",
        "homeworld": "http://swapi.dev/api/planets/21/",
        "mass": "unknown",
        "name": "Wilhuff Tarkin",
        "skin_color": "fair",
        "species": Array [],
        "starships": Array [],
        "url": "http://swapi.dev/api/people/12/",
        "vehicles": Array [],
      },
    ],
  },
  "type": "FILL_FILMS",
}
`;

exports[`Films actions should set fetching error 1`] = `
Object {
  "error": true,
  "payload": "We have an error !",
  "type": "SET_FETCHING_FILMS_ERROR",
}
`;

exports[`Films actions should start fetching 1`] = `
Object {
  "type": "START_FETCHING_FILMS",
}
`;

exports[`Films actions should stop fetching 1`] = `
Object {
  "type": "STOP_FETCHING_FILMS",
}
`;

exports[`People actions should fetch people 1`] = `[Function]`;

exports[`People actions should fill people 1`] = `
Object {
  "payload": Object {
    "count": 82,
    "results": Array [
      Object {
        "birth_year": "64BBY",
        "created": "2014-12-10T16:26:56.138000Z",
        "edited": "2014-12-20T21:17:50.330000Z",
        "eye_color": "blue",
        "films": Array [
          "http://swapi.dev/api/films/1/",
          "http://swapi.dev/api/films/6/",
        ],
        "gender": "male",
        "hair_color": "auburn, grey",
        "height": "180",
        "homeworld": "http://swapi.dev/api/planets/21/",
        "mass": "unknown",
        "name": "Wilhuff Tarkin",
        "skin_color": "fair",
        "species": Array [],
        "starships": Array [],
        "url": "http://swapi.dev/api/people/12/",
        "vehicles": Array [],
      },
    ],
  },
  "type": "FILL_FILMS",
}
`;

exports[`People actions should set fetching error 1`] = `
Object {
  "error": true,
  "payload": "We have an error !",
  "type": "SET_FETCHING_FILMS_ERROR",
}
`;

exports[`People actions should start fetching 1`] = `
Object {
  "type": "START_FETCHING_FILMS",
}
`;

exports[`People actions should stop fetching 1`] = `
Object {
  "type": "STOP_FETCHING_FILMS",
}
`;

exports[`Person actions should fetch person 1`] = `[Function]`;

exports[`Person actions should set fetching error 1`] = `
Object {
  "error": true,
  "payload": "We have an error !",
  "type": "SET_FETCHING_FILM_ERROR",
}
`;

exports[`Person actions should start fetching 1`] = `
Object {
  "type": "START_FETCHING_FILM",
}
`;

exports[`Person actions should stop fetching 1`] = `
Object {
  "type": "STOP_FETCHING_FILM",
}
`;
